### PR TITLE
fix: custom network settings form

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- Should add a scroll for custom network settings form, to avoid overflow of elements in the screen
- Should add a padding to keep the field on screen when the keyboard is open over the field
- Should add form validation for wallet-service fields, to avoid errors on saga processing
  - If any of the wallet-service fields are filled, then all the fields should be filled, otherwise an invalid error should appear in the empty field
  - if all the wallet-service fields are filled, or none are filled, then the form is allowed to be sent

It resolves: https://github.com/HathorNetwork/hathor-wallet-mobile/issues/458

iOS

https://github.com/HathorNetwork/hathor-wallet-mobile/assets/5992210/f861f1a2-e26a-4754-85a3-25c729c60666

Android

[custom-network-settings-android.webm](https://github.com/HathorNetwork/hathor-wallet-mobile/assets/5992210/7b775922-2068-4a87-9c98-f0af045e4e47)

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/screens/NetworkSettings/CustomNetworkSettingsScreen.js [hunk 1]
```diff
@@ -167,104 +181,115 @@ export const CustomNetworkSettingsScreen = ({ navigation }) => {
...
-        onBackPress={() => navigation.goBack()}
-      />
-
-      {isLoading(networkSettingsStatus) && (
-        <FeedbackModal
-          icon={<Spinner />}
-          text={feedbackLoadingText}
+    <KeyboardAvoidingView
+      behavior='padding'
+      style={{ flex: 1 }}
```
Comment (andreabadesso) [inline src/screens/NetworkSettings/CustomNetworkSettingsScreen.js:186]: Add to styles

File: src/screens/NetworkSettings/CustomNetworkSettingsScreen.js [hunk 2]
```diff
@@ -167,104 +181,115 @@ export const CustomNetworkSettingsScreen = ({ navigation }) => {
...
+        )}
 
-      {hasFailed(networkSettingsStatus) && (
-        <FeedbackModal
-          icon={(<Image source={errorIcon} style={styles.feedbackModalIcon} resizeMode='contain' />)}
-          text={feedbackFailedText}
-          onDismiss={handleFeedbackModalDismiss}
-        />
-      )}
+        {hasSucceed(networkSettingsStatus) && (
```
Comment (andreabadesso) [inline src/screens/NetworkSettings/CustomNetworkSettingsScreen.js:202]:
```suggestion
        {hasSucceeded(networkSettingsStatus) && (
```

File: src/screens/NetworkSettings/CustomNetworkSettingsScreen.js [hunk 3]
```diff
@@ -167,104 +181,115 @@ export const CustomNetworkSettingsScreen = ({ navigation }) => {
...
+          <FeedbackModal
+            icon={(<Image source={checkIcon} style={styles.feedbackModalIcon} resizeMode='contain' />)}
+            text={feedbackSucceedText}
```
Comment (andreabadesso) [inline src/screens/NetworkSettings/CustomNetworkSettingsScreen.js:205]:
```suggestion
            text={feedbackSucceededText}
```

**Review by andreabadesso — CHANGES_REQUESTED** [review]

**Review by pedroferreira1 — APPROVED** [review]

File: src/screens/NetworkSettings/CustomNetworkSettingsScreen.js (hunk 2 above)
Comment (alexruzenhack) [reply src/screens/NetworkSettings/CustomNetworkSettingsScreen.js]: Fixed

File: src/screens/NetworkSettings/CustomNetworkSettingsScreen.js (hunk 1 above)
Comment (alexruzenhack) [reply src/screens/NetworkSettings/CustomNetworkSettingsScreen.js]: Added

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**